(Microverse Portfolio) HTML, CSS, Linters & Readme

VelzckC0D3 opened the pull request:
# My Portfolio
- HTML, CSS & Linters properly working.
- I double-checked the linters and the gitignore, followed the requested tasks for the design template #2 

![Linters](https://user-images.githubusercontent.com/92229666/219433126-ec1aca20-9c71-4927-8f97-9fb3a4f99769.gif)
![image](https://user-images.githubusercontent.com/92229666/219458710-25f2e390-8bbf-4656-be7c-4021cfee6949.png)



# What have I submitted
- All the HTML checking that it follows a proper structuring as requested.
- All the CSS following the template #2 on the proposals given, created a different folder for the Icons and Images inside of style.  Using SVG to avoid loss of images or icons quality
- Checked that all the Linters properly worked before submitting the pull request.
- Prepared a personalized README.MD file to describe what's going on in the project and explain its functionalities.

# (Corrections 1st Code Review Request)♻️
✅Corrected the logo so now it redirects you to the main section (header) according to the best practices. 
✅Added "nav" tag to the "hamburger" (menu) button.
✅Putted all the Icons inside an ul / li tag.
✅Background image properly displayed now!
✅Also, added on this Pull Request everything that I have added to the Repository.

# New! (Corrections 2nd Code Review Request)♻️ I'm scared 😱
## ✅Corrected the background ⬇️ so now it's established on `#main` with CSS instead of being a div on the HTML index.
![image](https://user-images.githubusercontent.com/92229666/219495954-2cda4da5-bbed-4c6e-abee-d249682993b0.png)
## ✅Corrected the `style/icon/**iconname**` ⬇️, so it matches the HTML file 
![image](https://user-images.githubusercontent.com/92229666/219497452-f2e190f7-0822-4129-9da1-b0eeabbd8547.png)



# Thanks for your review! (again x2)  Please, have mercy! 🥺
![thumbs up](https://user-images.githubusercontent.com/92229666/219434040-db557c22-9dd2-4fe8-9d5a-3e17cd213b64.gif)

VelzckC0D3 committed:
HTML & Linters

Added the basic HTML and the functional Linters to my portfolio project, also checked that .gitignore was properly working on the node_modules folder.

VelzckC0D3 committed:
CSS & Images

Added the CSS, the background, the images and the icons to the project. I picked SVG because it offers a better quality for the project

VelzckC0D3 committed:
Stylelint corrections

Some corrections that Stylelint requested me for the style.css file

VelzckC0D3 committed:
Readme.md

Added the personalized Readme file in order to show what the project is offering, also added a Live Demo so the user can see the website.

Corrected minor style.css conflicts with the Icons and the BackGround

VelzckC0D3 committed:
Update Readme.md

Added Gift Animations and the actual Live Demo to the readme, all descriptions corrected.

ABDULALI3468 commented on index.html:
```diff
@@ -1,15 +1,61 @@
+
+        <div id="header" class="flexColumn">
+            <div id="headerNav" class="headerCont roboto">
+                <p id="hour">VelzckC0D3</p>
```
Please consider adding a (`a` tag) to your logo.
    - This is for three reasons: convention, semantics, and user experience(UX). [Check this link](https://ux.stackexchange.com/questions/81727/why-is-it-standard-for-a-website-logo-to-navigate-to-the-home-page) for more info.

VelzckC0D3 replied:
Thanks! I'll do it right away
![image](https://user-images.githubusercontent.com/92229666/219451756-813fee59-77a9-4130-be5d-d717276679ab.png)

ABDULALI3468 commented on index.html:
```diff
@@ -1,15 +1,61 @@
+        <div id="header" class="flexColumn">
+            <div id="headerNav" class="headerCont roboto">
+                <p id="hour">VelzckC0D3</p>
+                <img id="menu" src="style/img/menu.svg" alt="">
```
Please consider adding a  `nav` tag to your HAMBURGER icon.
    - The `nav` tag is for semantics reasons, and the right position for convention and user experience for mobile devices.

VelzckC0D3 replied:
Done, thank you for your correction! appreciated

VelzckC0D3 replied:
Done, thank you for your correction! appreciated
![image](https://user-images.githubusercontent.com/92229666/219451525-2363d2fc-4beb-4e2b-9943-9b8c7825f4c8.png)

ABDULALI3468 commented on index.html:
```diff
@@ -1,15 +1,61 @@
+
+    </header>
+
+    <main class="flexColumn">
```
Please consider adding the background image to the entire section using CSS property as presented in the [Figma Template](https://www.figma.com/file/l7SqJ3ZfkAKih9sFxvWSR4/Microverse-Student-Project-1?node-id=1%3A1471&t=NaDEEccFhkBWgr7P-0) so it looks more professional and a look alike to the design you have chosen to create your portfolio.

ABDULALI3468 commented on index.html:
```diff
@@ -1,15 +1,61 @@
+                <p>I can help you build a product , feature or website Look through some of my work and experience! If
+                    you like what you see and have a project you need coded,
+                    don’t hestiate to contact me.</p>
+                <div id="mainIcons">
+                    <a href="https://github.com/VelzckC0D3/" target="_blank" rel="noopener" aria-label="Github"><img class="icon" src="style/icons/github.svg"
+                            alt="GitHub"></a>
+                    <a href="https://www.linkedin.com/in/velzckcode/" target="_blank" rel="noopener" aria-label="Linkedin"> <img class="icon" src="style/icons/linkedin.svg"
+                            alt="Linkedin"></a>
+                    <a href="https://www.angellist.com/" target="_blank" rel="noopener" aria-label="AngelList"> <img class="icon" src="style/icons/angelList.svg"
+                            alt="AngelList"></a>
+                    <a href="https://twitter.com/velzck/" target="_blank" rel="noopener" aria-label="Twitter"> <img class="icon" src="style/icons/twitter.svg"
+                            alt="Twitter"></a>
+                    <a href="https://medium.com/" target="_blank" rel="noopener" aria-label="Medium"> <img class="icon" src="style/icons/medium.svg" alt=""></a>
+
+                </div>
+            </div>
```
The list of icons needs to have a `ul tag` to group them, and `li tag` per icon; each icon has a link (a tag).

Apart from being semantically correct, the `a tag` will let you connect your social networks with the icons, using `href attribute`.

VelzckC0D3 replied:
Done! 
![image](https://user-images.githubusercontent.com/92229666/219451647-1d324f87-c20b-4730-8e42-dd109f00e3ae.png)

VelzckC0D3 replied:
# Also, added on the PR the specifications of everything that I had added, as you requested! 
![image](https://user-images.githubusercontent.com/92229666/219452914-4fba0aa2-308a-4fe0-bc58-225455c94092.png)

## And the background image is properly added now! 
![image](https://user-images.githubusercontent.com/92229666/219462543-d029972f-e5af-49cb-a0bf-c40676124e79.png)

ABDULALI3468 reviewed (changes requested):
# Changes Required 👍 

Hi @VelzckC0D3 , 

<img src="https://media1.giphy.com/media/31Q36EHJ7C5ZoIMw79/giphy.gif"/>

Indeed you did an exceptional job but still, there are some changes that need to be addressed to go to the next level 💯


## Highlights 
✅All linters are checked.
✅Project is professionally documented.
✅The right GitHub flow method has been used.
✅Professional pull request title and description.


 ## Required Changes ♻️

Please consider making your PR description tell what you have done in your PR. You can consider the following PR for an [example.](https://github.com/ABDULALI3468/bookMe-frontend/pull/10)

> **_Check the comments below to have a look at the required changes._**

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.

> * This is My **GitHub**: ![GitHub followers](https://img.shields.io/github/followers/ABDULALI3468?label=ABDULALI&style=social)
> * This is My **Linkedin**: [![Linkedin Badge](https://img.shields.io/badge/-ABDUL%20ALI-blue?style=flat-square&logo=Linkedin&logoColor=white&link=https://www.linkedin.com/in/abdul-ali-5400bb216/)](https://www.linkedin.com/in/abdul-ali-5400bb216/)&nbsp;
> * This is My **Twitter**: [![Twitter Badge](https://img.shields.io/badge/-@mrabdul_ali_-1ca0f1?style=flat-square&labelColor=1ca0f1&logo=twitter&logoColor=white&link=https://twitter.com/mrabdul_ali)](https://twitter.com/mrabdul_ali)&nbsp;

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VelzckC0D3 committed:
Code Request Review Corrections

As my Code Reviewer requested, made the changes so my website follows the best practices

First, I updated the logo so it redirect you to the main (header) section of the website, also added a nav tag to the "hamburger" (menu) icon, then putted all the Icons inside an "ul" tag, each one with his own "li" container.

added all the committed changes to the readme file.

amejid commented on index.html:
```diff
@@ -1,15 +1,70 @@
+                                    class="icon" src="style/icons/medium.svg" alt=""></a></li>
+                    </ul>
+                </div>
+                <img id="BG" src="style/BG.svg" alt="BackGround">
```
- [ ] Great job with the design. Let's fix some small things to achieve the project's objective. This image here is only included for decorative reasons and is not part of the contents of the page. Kindly remove it from here and add it as a `background-image` using CSS as suggested by the previous reviewer. 🤝️ You can refer to this [Stackoverflow](https://stackoverflow.com/questions/492809/when-to-use-img-vs-css-background-image) discussion to get more insight about the topic.

VelzckC0D3 replied:
# Done!! I'm really scared at this point 😖
![image](https://user-images.githubusercontent.com/92229666/219485763-18efbf46-0341-420c-af39-35ad47bdc9f2.png)
## Thank you for the feedback, Amejid ! now because of this change, the background of the website is fully responsive to any width 🙌

amejid commented on index.html:
```diff
@@ -1,15 +1,70 @@
+                    don’t hestiate to contact me.</p>
+                <div>
+                    <ul id="mainIcons">
+                        <li> <a href="https://github.com/VelzckC0D3/" target="_blank" rel="noopener"
+                                aria-label="Github"><img class="icon" src="style/icons/github.svg" alt="GitHub"></a>
+                        </li>
+                        <li> <a href="https://www.linkedin.com/in/velzckcode/" target="_blank" rel="noopener"
+                                aria-label="Linkedin"> <img class="icon" src="style/icons/linkedin.svg"
+                                    alt="Linkedin"></a></li>
+                        <li> <a href="https://www.angellist.com/" target="_blank" rel="noopener" aria-label="AngelList">
+                                <img class="icon" src="style/icons/angelList.svg" alt="AngelList"></a></li>
+                        <li> <a href="https://twitter.com/velzck/" target="_blank" rel="noopener" aria-label="Twitter">
+                                <img class="icon" src="style/icons/twitter.svg" alt="Twitter"></a></li>
+                        <li> <a href="https://medium.com/" target="_blank" rel="noopener" aria-label="Medium"> <img
+                                    class="icon" src="style/icons/medium.svg" alt=""></a></li>
```
- [ ] All the icons you are using here are broken. I think this is happening because of the file names. All the images inside your folder start with uppercase letters while in the code you are referencing them using lowercase letters. This is how it is currently looking. Kindly use the correct file names to address this issue. 😎️

<img src="https://user-images.githubusercontent.com/17243325/219465496-f65d3866-9e48-4d99-9735-3a8640726ef9.png" width="300">

VelzckC0D3 replied:
## Also, done! I had to change the folder name in order to see the changes reflected on the repository 
![image](https://user-images.githubusercontent.com/92229666/219486578-3cbcd4a7-60b2-43e6-822c-fbc9dba84d6b.png)
![image](https://user-images.githubusercontent.com/92229666/219486867-ccac92f1-a963-40f6-9fe6-cd2f4a80a945.png)

amejid reviewed (changes requested):
Hi @VelzckC0D3, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

<img src="https://media2.giphy.com/media/t8YPf3ACKeouLzEIvr/giphy.gif"/>

 ## Highlights
- Linter checks passing :heavy_check_mark: 
- Descriptive PR :heavy_check_mark: 
- Correct use of Github flow :heavy_check_mark: 
- Nice UI design :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

VelzckC0D3 committed:
Last Commit Final Corrections

As requsted, I deleted the Background image of HTML and added directly to the main tag on CSS, also corrected the names of the icons (github.svg, linkedin.svg, angellist.svg, twitter.svg and medium.svg) to all lowercase and double checked on the folder that it was matching.

VelzckC0D3 committed:
Updated icon names

Updated the icon names to lowercase so I can avoid any inconvenience with the HTML file

VelzckC0D3 committed:
Reponsve & Background Corrections

Finished details for the background so it properly adjusts to any screen, also edited the max height of main so it doesn't have any visual bug on wider displays

VelzckC0D3 committed:
Move BG to img folder

Moved BG.svg to IMG folder for more optimization, properly positioned, also hide the scrollbar for desktop devices.

VelzckC0D3 committed:
Last Details

I finished to polish things on the responsive design, such as max widths on the H1 and P of the main section, and also hide the overflow X for the Main section in order to avoid inconveniences with the viewport

Shisui6 commented on README.md:
```diff
@@ -1 +1,210 @@
+
+This project is [MIT](./LICENSE) licensed.
+
+_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
```
[Optional] Kindly remove line 208 from your readme file. It is only an instruction to help you know what to do in this section and should not be part of the final readme😎

Shisui6 reviewed (approved):
Hi @VelzckC0D3 , 

# Approved 🪄
Your project is complete! There is nothing else to say other than... it's time to merge it 💎
Congratulations! 🎉 

## Highlights
- Great design ✅
- Good use of flexbox ✅

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code. Some of them were simply missed by the previous reviewer, and addressing them will really improve your application._

You can also consider the following:

N/A

Cheers and Happy coding!🍻💻

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561), you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
